test/transport/basic.lean

```diff
@@ -0,0 +1,85 @@
+import tactic.transport_2
+import order.bounded_lattice
+import algebra.lie_algebra
+
+-- We verify that `transport` can move a `semiring` across an equivalence.
+-- Note that we've never even mentioned the idea of addition or multiplication to `transport`.
+def semiring.map {α : Type} [semiring α] {β : Type} (e : α ≃ β) : semiring β :=
+by transport
+.
+
+-- Indeed, it can equally well move a `semilattice_sup_top`.
+def sup_top.map {α : Type} [semilattice_sup_top α] {β : Type} (e : α ≃ β) : semilattice_sup_top β :=
+by transport
+.
+
+def lie_ring.map {α : Type} [lie_ring α] {β : Type} (e : α ≃ β) : lie_ring β :=
+by transport.
+
+-- Below we verify that the transported structure is definitionally what you would hope for.
+
+-- FIXME
+-- In fact, as defined in semiring.map above, it's _not_ definitionally what you want. :-(
+-- However just taking the definition of `semiring.map` produced `by transport` above,
+-- and simplifying it, gives something that _is_ definitionally what you want!
+-- Surely it's possible to arrange this without making a new declaration...
+
+simp_defn semiring.map semiring.map'
```

Comment:
I would love some help understanding how to avoid this.

The problem here is that `semiring.map` itself is definitionally gross. However, if we make a copy of the declaration, simplifying the body, everything works great: see below.

How do we just incorporate this simplification step into the tactic itself?